src/cholesky/lkjcholesky.jl

```diff
@@ -136,7 +136,7 @@ function logkernel(d::LKJCholesky, R::LinearAlgebra.Cholesky)
 end
 
 function logpdf(d::LKJCholesky, R::LinearAlgebra.Cholesky)
-    insupport(d, R) || throw(ArgumentError("provided point is not in the support"))
+    insupport(d, R) || return oftype(_logpdf(d, LinearAlgebra.Cholesky(zeros(eltype(R.factors), d.d, d.d), R.uplo, R.info)), -Inf)
     return _logpdf(d, R)
```

Comment:
The only problem I can think of here is that it does not necessarily preserve the type of `R.factors` (e.g., if it has some special structure or is some special matrix type).

It's not clear to me though why constructing a new `Cholesky` is needed here at all. `_logpdf` just sums `logkernel` and the `logc0` constant (independent of the input), and the `logkernel` function just above should work (or at least not error) even if `Cholesky` has the wrong size. The only problem there would be a `Cholesky` with negative values on the diagonal of `R.factors`, and that should not be possible (if one does not circumvent the error checks at construction time).

So I think maybe (untested) we could just use
```suggestion
    lp = _logpdf(d, R)
    return insupport(d, R) ? lp : oftype(lp, -Inf) 
```

Reply:
Done. It seems to work fine.